pkg/api/pluginproxy/token_cache.go

```diff
@@ -0,0 +1,124 @@
+package pluginproxy
+
+import (
+	"context"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+type TokenCredential interface {
+	GetCacheKey() string
+	GetAccessToken(ctx context.Context, scopes []string) (*azcore.AccessToken, error)
```

Comment:
...or, create your own type for AccesstToken and make that the return type instead of azcore.AccessToken.

Reply:
Created